src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -413,22 +434,105 @@ def save_tile(ix, iy, das):
         # Run in a separate thread
         threading.Thread(target=save_tile, args=(ix, iy, das), ).start()
 
-    def _acquireTile(self, i, ix, iy):
+    def _constructCube(self, images):
+        """
+        Construct xyz cube from a  z stack of images
+        :param images: (list) list of z ordered images
+        :return: (DataArray) the da of the xyz cube
+        """
+        # images is a list of 3 dim data arrays.
+        ret = []
+        for image in images:
+            stack = numpy.dstack(image)
+            stack = numpy.swapaxes(stack, 1, 2)
+            ret.append(stack[0])
+
+        # Add back metadata
+        metadata3d = copy.copy(images[0].metadata)
+        # Extend pixel size to 3D
+        ps_x, ps_y = metadata3d[model.MD_PIXEL_SIZE]
+        ps_z = ZSTEP
+
+        # Computer cube centre
+        c_x, c_y = metadata3d[model.MD_POS]
+        c_z = self._zlevels[len(self._zlevels)//2]  # Assuming zlevels are ordered
```

Comment:
Let's use the center of zlevels (it's going to be different when there is an even number): `(zlevels[0] + zlevels[-1]) / 2`